reflex/vars/datetime.py

```diff
@@ -0,0 +1,184 @@
+"""Immutable datetime and date vars."""
+
+from __future__ import annotations
+
+from datetime import date, datetime
+from typing import Any, NoReturn, TypeVar, Union, overload
+
+from reflex.utils.exceptions import VarTypeError
+from reflex.vars.number import BooleanVar
+
+from .base import CustomVarOperationReturn, Var, var_operation, var_operation_return
+
+DATETIME_T = TypeVar("DATETIME_T", datetime, date)
+
+datetime_types = Union[datetime, date]
+
+
+def raise_var_type_error():
+    """Raise a VarTypeError.
+
+    Raises:
+        VarTypeError: Cannot compare a datetime object with a non-datetime object.
+    """
+    raise VarTypeError("Cannot compare a datetime object with a non-datetime object.")
+
+
+class DateTimeVar(Var[DATETIME_T], python_types=(datetime, date)):
```

Comment:
Something that is attempting to be of subclass `Var` should also provide a literal var that subclasses it.